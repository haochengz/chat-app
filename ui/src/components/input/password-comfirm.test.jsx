
import React from 'react'
import { shallow, mount } from 'enzyme'

import PasswordComfirmInput from './password-comfirm'

describe('<PasswordComfirmInput />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<PasswordComfirmInput />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should emit value to the parent component by password input', () => {
    let valueMock
    const inputMock = v => valueMock = v
    const wrapper = shallow(
      <PasswordComfirmInput
        onChange={inputMock}
      />
    )
    const input = wrapper.find({placeholder: 'Password'})
    input.simulate('change', 'test')
    expect(valueMock).toEqual('test')
  })

  it('should stores its own data of password repeated input', () => {
    let valueMock
    const inputMock = v => valueMock = v
    const wrapper = shallow(
      <PasswordComfirmInput
        onChange={inputMock}
      />
    )
    const input = wrapper.find({placeholder: 'Repeat the password'})
    input.simulate('change', 'test')
    expect(wrapper.state('__p')).toBe('test')
    expect(valueMock).toBe('')
  })

  it('should emit errors if there is an invalid value in the password input', () => {
    let errors = null
    function change(v, e) {
      errors = e
    }
    const wrapper = shallow(
      <PasswordComfirmInput
        onChange={change}
      />
    )
    const input = wrapper.find({placeholder: 'Password'})
    input.simulate('change', 'test')
    expect(errors).toHaveLength(2)
  })

  it('should emit errors then changing the repeat password input', () => {
    let errors = null
    function change(v, e) {
      errors = e
    }
    const wrapper = shallow(
      <PasswordComfirmInput
        onChange={change}
      />
    )
    const input = wrapper.find({placeholder: 'Repeat the password'})
    input.simulate('change', 'test')
    expect(errors).toHaveLength(3)
  })

  it('should only allow letters, numbers and underscore as the value', () => {
    let errors = null
    function change(v, e) {
      errors = e
    }
    const wrapper = shallow(
      <PasswordComfirmInput
        onChange={change}
      />
    )
    const input = wrapper.find({placeholder: 'Password'})
    input.simulate('change', 'test%test')
    expect(errors).toHaveLength(2)
  })

  it('should be ok if password entered correctly', () => {
    let errors = null
    function change(v, e) {
      errors = e
    }
    const wrapper = shallow(
      <PasswordComfirmInput
        onChange={change}
        value={'testtest'}
      />
    )
    const input = wrapper.find({placeholder: 'Repeat the password'})
    input.simulate('change', 'testtest')
    expect(errors).toHaveLength(0)
  })

  it('should render two error badges if any errors passing in to the component', () => {
    const wrapper = mount(
      <PasswordComfirmInput
        errors={['error1', 'error2']}
      />
    )
    expect(wrapper.exists('.am-input-error-extra')).toBe(true)
  })

  it('should no error badge if no errors passing in to the component', () => {
    const wrapper = mount(
      <PasswordComfirmInput
        errors={[]}
      />
    )
    expect(wrapper.exists('.am-input-error-extra')).toBe(false)
  })

  it('should calls the showErrors method if error badge had been clicked and there are errors in error array', () => {
    const errorHandler = jest.fn()
    const wrapper = mount(
      <PasswordComfirmInput
        errors={['error1', 'error2']}
      />
    )
    wrapper.instance().showErrors = errorHandler
    wrapper.instance().forceUpdate()
    const errorBadge = wrapper.find('.am-input-error-extra').at(0)
    errorBadge.simulate('click')
    expect(errorHandler).toBeCalled()
  })
})