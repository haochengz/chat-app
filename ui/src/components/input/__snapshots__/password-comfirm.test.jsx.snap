// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordComfirmInput /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordComfirmInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InputItem
          clear={true}
          disabled={false}
          editable={true}
          error={false}
          extra=""
          labelNumber={5}
          moneyKeyboardAlign="right"
          moneyKeyboardWrapProps={Object {}}
          onBlur={[Function]}
          onChange={[Function]}
          onErrorClick={[Function]}
          onExtraClick={[Function]}
          onFocus={[Function]}
          onVirtualKeyboardConfirm={[Function]}
          placeholder="Password"
          prefixCls="am-input"
          prefixListCls="am-list"
          type="password"
          updatePlaceholder={false}
          value={undefined}
>
          Password:
</InputItem>,
        <InputItem
          clear={true}
          disabled={false}
          editable={true}
          error={false}
          extra=""
          labelNumber={5}
          moneyKeyboardAlign="right"
          moneyKeyboardWrapProps={Object {}}
          onBlur={[Function]}
          onChange={[Function]}
          onErrorClick={[Function]}
          onExtraClick={[Function]}
          onFocus={[Function]}
          onVirtualKeyboardConfirm={[Function]}
          placeholder="Repeat the password"
          prefixCls="am-input"
          prefixListCls="am-list"
          type="password"
          updatePlaceholder={false}
          value=""
>
          Password:
</InputItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Password:",
          "clear": true,
          "disabled": false,
          "editable": true,
          "error": false,
          "extra": "",
          "labelNumber": 5,
          "moneyKeyboardAlign": "right",
          "moneyKeyboardWrapProps": Object {},
          "onBlur": [Function],
          "onChange": [Function],
          "onErrorClick": [Function],
          "onExtraClick": [Function],
          "onFocus": [Function],
          "onVirtualKeyboardConfirm": [Function],
          "placeholder": "Password",
          "prefixCls": "am-input",
          "prefixListCls": "am-list",
          "type": "password",
          "updatePlaceholder": false,
          "value": undefined,
        },
        "ref": null,
        "rendered": "Password:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Password:",
          "clear": true,
          "disabled": false,
          "editable": true,
          "error": false,
          "extra": "",
          "labelNumber": 5,
          "moneyKeyboardAlign": "right",
          "moneyKeyboardWrapProps": Object {},
          "onBlur": [Function],
          "onChange": [Function],
          "onErrorClick": [Function],
          "onExtraClick": [Function],
          "onFocus": [Function],
          "onVirtualKeyboardConfirm": [Function],
          "placeholder": "Repeat the password",
          "prefixCls": "am-input",
          "prefixListCls": "am-list",
          "type": "password",
          "updatePlaceholder": false,
          "value": "",
        },
        "ref": null,
        "rendered": "Password:",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InputItem
            clear={true}
            disabled={false}
            editable={true}
            error={false}
            extra=""
            labelNumber={5}
            moneyKeyboardAlign="right"
            moneyKeyboardWrapProps={Object {}}
            onBlur={[Function]}
            onChange={[Function]}
            onErrorClick={[Function]}
            onExtraClick={[Function]}
            onFocus={[Function]}
            onVirtualKeyboardConfirm={[Function]}
            placeholder="Password"
            prefixCls="am-input"
            prefixListCls="am-list"
            type="password"
            updatePlaceholder={false}
            value={undefined}
>
            Password:
</InputItem>,
          <InputItem
            clear={true}
            disabled={false}
            editable={true}
            error={false}
            extra=""
            labelNumber={5}
            moneyKeyboardAlign="right"
            moneyKeyboardWrapProps={Object {}}
            onBlur={[Function]}
            onChange={[Function]}
            onErrorClick={[Function]}
            onExtraClick={[Function]}
            onFocus={[Function]}
            onVirtualKeyboardConfirm={[Function]}
            placeholder="Repeat the password"
            prefixCls="am-input"
            prefixListCls="am-list"
            type="password"
            updatePlaceholder={false}
            value=""
>
            Password:
</InputItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Password:",
            "clear": true,
            "disabled": false,
            "editable": true,
            "error": false,
            "extra": "",
            "labelNumber": 5,
            "moneyKeyboardAlign": "right",
            "moneyKeyboardWrapProps": Object {},
            "onBlur": [Function],
            "onChange": [Function],
            "onErrorClick": [Function],
            "onExtraClick": [Function],
            "onFocus": [Function],
            "onVirtualKeyboardConfirm": [Function],
            "placeholder": "Password",
            "prefixCls": "am-input",
            "prefixListCls": "am-list",
            "type": "password",
            "updatePlaceholder": false,
            "value": undefined,
          },
          "ref": null,
          "rendered": "Password:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Password:",
            "clear": true,
            "disabled": false,
            "editable": true,
            "error": false,
            "extra": "",
            "labelNumber": 5,
            "moneyKeyboardAlign": "right",
            "moneyKeyboardWrapProps": Object {},
            "onBlur": [Function],
            "onChange": [Function],
            "onErrorClick": [Function],
            "onExtraClick": [Function],
            "onFocus": [Function],
            "onVirtualKeyboardConfirm": [Function],
            "placeholder": "Repeat the password",
            "prefixCls": "am-input",
            "prefixListCls": "am-list",
            "type": "password",
            "updatePlaceholder": false,
            "value": "",
          },
          "ref": null,
          "rendered": "Password:",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
